pkg/monitors/pingdom/pingdom-monitor.go

```diff
@@ -82,6 +84,26 @@ func (service *PingdomMonitorService) GetAll() []models.Monitor {
 	return monitors
 }
 
+// Function that gets all monitors with an error handling to avoid duplicate checks creation
+func (service *PingdomMonitorService) GetAllWithAPIErrorHandler() ([]models.Monitor, error) {
```

Comment:
I don't know how this function can avoid duplication compared with the existing Getall() one?